### PR TITLE
docs: update dependencies

### LINKED ISSUE #2975: Fix docs toolchain
Right now we'll pin it, but we should upgrade to sphinx 7.

### PR DESCRIPTION
**Pull request by agoose77**
Closes #2975

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-jupyterlite-pyodide-kernel==0.1.0
-jupyterlite[piplite]==0.1.0
-jupyterlite-sphinx==0.8.0
+# This kernel requires a specific pyodide version (see pyodide-build)
+jupyterlite-pyodide-kernel==0.2.*
+jupyterlite[piplite]==0.2.*
+jupyterlite-sphinx==0.11.*
 
-sphinx>=4.5.0,<5.0.0
+sphinx==7.*
```
Comment (jpivarski) [inline docs/requirements.txt]: Pin to a version? The [admonitions](https://iscinumpy.dev/post/bound-version-constraints/) against `<` must apply to `=` as well[^1]; is there a reason to think that Sphinx 8 would be bad?

[^1]: If they don't, then I don't understand the issue.

File: docs/requirements-wasm.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-pyodide-build==0.23.3
-pydantic<2
+# This determines the pyodide version
+pyodide-build==0.24.1
```
Comment (jpivarski) [inline docs/requirements-wasm.txt]: I can be convinced that an exact version of Pyodide is needed, since extensions are vendored into it.

**Review by jpivarski — APPROVED** [review]
I'm in favor of any docs-tech updates that you think are appropriate.

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,13 +1,15 @@
 # JupyterLite (kernel requires specific pyodide version)
-jupyterlite-pyodide-kernel==0.1.0
-jupyterlite[piplite]==0.1.0
-jupyterlite-sphinx==0.8.0
+# This kernel requires a specific pyodide version (see pyodide-build)
+jupyterlite-pyodide-kernel==0.2.*
+jupyterlite[piplite]==0.2.*
+jupyterlite-sphinx==0.11.*
```
Comment (jpivarski) [inline docs/requirements.txt]: Same argument as for Pyodide.

File: docs/requirements.txt (hunk 1 above)
Comment (jpivarski) [reply docs/requirements.txt]: > Right now we'll pin it, but we should upgrade to sphinx 7.

Okay!

Comment (agoose77) [reply docs/requirements.txt]: For application-like uses, we can be very restrictive e.g. pin exactly. I'm not in favour of going _that_ far; we just want our docs to break less often.

File: docs/requirements.txt (hunk 3 above)
Comment (agoose77) [reply docs/requirements.txt]: My assumption is that jupyterlite-pyodide-kernel will receive a minor bump if the pyodide version changes.

**Comment by agoose77** [comment]
Looks like pyodide broke the emscripten toolchain bundling in this version, but jupyterlite doesn't have a kernel that supports the newer version yet. I'll update this PR to upgrade only Sphinx

**Comment by agoose77** [comment]
This needs more work; I'm still recovering my development environment, but will finalise this PR soon.

**Comment by jpivarski** [comment]
I decided to postpone it for after 2.6.0. I merged what was mergeable, and have now started the awkward-cpp build.

**Comment by agoose77** [comment]
I've changed this PR to ignore the jupyterlite changes, and just touch the sphinx parts. Let's see what happens.

**Pull request merged**